### PR TITLE
[NOD-530] Remove coinbase inputs and add blue score to payload

### PR DESCRIPTION
**Pull request by someone235**

### CODE REVIEW

File: blockdag/utxoset.go [hunk 1]
```diff
@@ -457,17 +457,15 @@ func (fus *FullUTXOSet) WithDiff(other *UTXODiff) (UTXOSet, error) {
...
+	}
 
-		for _, txIn := range tx.TxIn {
-			fus.remove(txIn.PreviousOutpoint)
-		}
+	for _, txIn := range tx.TxIn {
+		fus.remove(txIn.PreviousOutpoint)
 	}
 
+	isCoinbase := tx.IsCoinBase()
```
Comment (svarogg) [inline blockdag/utxoset.go:468]: Here too

File: util/block.go [hunk 2]
```diff
@@ -38,7 +39,7 @@ type Block struct {
 	blockHash       *daghash.Hash  // Cached block hash
 	transactions    []*Tx          // Transactions
 	txnsGenerated   bool           // ALL wrapped transactions generated
-	blueScore       uint64         // Blue score
+	blueScore       *uint64        // Blue score
```
Comment (svarogg) [inline util/block.go:42]: Please, unless you already done that, make sure nobody is using blueScore directly.
Add to comment about Blue score that it shouldn't be used, and use .BlueScore instead

File: util/coinbasepayload/coinbasepayload.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	if err != nil {
+		return nil, err
+	}
+	return w.Bytes(), nil
+}
+
+// ErrIncorrectScriptPubKeyLen indicates that the script pub key length is not as expected.
+var ErrIncorrectScriptPubKeyLen = errors.New("incorrect script pub key length")
+
+// DeserializeCoinbasePayload deserialize the coinbase payload to its component (scriptPubKey and extra data).
```
Comment (svarogg) [inline util/coinbasepayload/coinbasepayload.go:38]: typo: DeserializeCoinbasePayload deserialize**s**

File: blockdag/dag.go [hunk 4]
```diff
@@ -658,22 +658,20 @@ func (node *blockNode) selectedParentMultiset(dag *BlockDAG) (*secp256k1.MultiSe
...
 
 func addTxToMultiset(ms *secp256k1.MultiSet, tx *wire.MsgTx, pastUTXO UTXOSet, blockBlueScore uint64) (*secp256k1.MultiSet, error) {
-	isCoinbase := tx.IsCoinBase()
-	if !isCoinbase {
-		for _, txIn := range tx.TxIn {
-			entry, ok := pastUTXO.Get(txIn.PreviousOutpoint)
-			if !ok {
-				return nil, errors.Errorf("Couldn't find entry for outpoint %s", txIn.PreviousOutpoint)
-			}
+	for _, txIn := range tx.TxIn {
```
[the remark this answers is not in the document]
Comment (someone235) [reply blockdag/dag.go:661]: Yes, it's a transaction from prev block, and we validate it in validateCoinbaseTransaction

File: blockdag/dag.go [hunk 5]
```diff
@@ -658,22 +658,20 @@ func (node *blockNode) selectedParentMultiset(dag *BlockDAG) (*secp256k1.MultiSe
...
-				return nil, err
-			}
+		var err error
+		ms, err = removeUTXOFromMultiset(ms, entry, &txIn.PreviousOutpoint)
+		if err != nil {
+			return nil, err
 		}
 	}
 
+	isCoinbase := tx.IsCoinBase()
```
[the remark this answers is not in the document]
Comment (someone235) [reply blockdag/dag.go:674]: It's used in a loop

File: blockdag/utxoset.go (hunk 1 above)
Comment (someone235) [reply blockdag/utxoset.go:468]: Here too

**Comment by someone235** [comment]
Rebuild after update in master

**Comment by someone235** [comment]
Rebuild after update in master

**Pull request merged**